pandas/core/indexes/datetimelike.py

```diff
@@ -168,36 +169,43 @@ def __contains__(self, key: Any) -> bool:
             is_scalar(res) or isinstance(res, slice) or (is_list_like(res) and len(res))
         )
 
-    def sort_values(self, return_indexer=False, ascending=True):
+    def sort_values(self, return_indexer=False, ascending=True, key=None):
         """
         Return sorted copy of Index.
         """
+        idx = ensure_key_mapped(self, key)
+
+        _as = idx.argsort()
+        if not ascending:
+            _as = _as[::-1]
+        sorted_index = self.take(_as)
+
         if return_indexer:
-            _as = self.argsort()
-            if not ascending:
-                _as = _as[::-1]
-            sorted_index = self.take(_as)
             return sorted_index, _as
         else:
```

Comment:
@jreback do you know what this second branch is here for? I'm tempted to make the `argsort` method the default since it's necessary for `key` sorting, unless it's an important optimization.